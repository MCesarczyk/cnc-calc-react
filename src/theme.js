export const theme = {
    color: {
        primary: "#1e3246",
        background: "#133246",
    },
    breakpoint: {
        mobileMax: "480px",
    },
}