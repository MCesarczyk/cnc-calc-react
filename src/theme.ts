export const theme = {
  color: {
    primary: '#1e3246',
    primaryLight: '#2e4c6b',
    primaryText: '#000',
    contrastText: '#fff',
    background: '#133246',
    elementBackground: '#fff',
    submitButtonText: '#19aa6e',
    resetButtonText: '#dc3223',
    copyButtonText: '#dc143c',
    activeNavlinkText: '#2188FF',
    invalid: 'beige',
    invalidBackground: 'beige',
    disabled: '#ddd',
    disabledText: '#808080',
    disabledBackground: '#ddd',
    placeholder: '#747474',
  },
  breakpoint: {
    xs: '480px',
    sm: '576px',
    md: '768px',
    lg: '992px',
    xl: '1200px',
    xxl: '1400px',
  },
};
