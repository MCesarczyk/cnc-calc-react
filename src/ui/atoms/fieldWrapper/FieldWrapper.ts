import styled from 'styled-components';

export type FieldWrapperVariants = 'DEFAULT';

export type Palette<TVariants extends string, TStyles> = {
  [variant in TVariants]: TStyles;
};

interface FieldWrapperProps {
  $variant?: FieldWrapperVariants;
}

export const FieldWrapper = styled.div<FieldWrapperProps>`
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;

  @media (max-width: ${({ theme }) => theme.breakpoint.xs}) {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
`;

FieldWrapper.defaultProps = {
  $variant: 'DEFAULT',
  children: 'Sample content',
};

FieldWrapper.displayName = 'FieldWrapper';
