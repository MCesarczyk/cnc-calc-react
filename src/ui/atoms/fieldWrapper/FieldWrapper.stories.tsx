import { ComponentProps } from 'react';

import type { Meta, StoryFn } from '@storybook/react';

import { FieldWrapper } from './FieldWrapper';

const meta = {
  title: 'Atoms/FieldWrapper',
  component: FieldWrapper,
  parameters: {
    layout: 'centered',
  },
  tags: ['autodocs'],
  argTypes: {
    variant: {
      table: {
        disable: true,
      },
    },
  },
} satisfies Meta<typeof FieldWrapper>;

export default meta;

const Template: StoryFn<ComponentProps<typeof FieldWrapper>> = (args) => <FieldWrapper { ...args } />;

export const Default = Template.bind({});
Default.args = {
  $variant: 'DEFAULT',
};
