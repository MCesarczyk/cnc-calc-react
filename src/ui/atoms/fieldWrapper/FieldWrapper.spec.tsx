import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import { theme } from 'theme';
import { FieldWrapper } from 'ui/atoms/fieldWrapper';

test('Text sub rendered properly', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <FieldWrapper />
    </ThemeProvider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
