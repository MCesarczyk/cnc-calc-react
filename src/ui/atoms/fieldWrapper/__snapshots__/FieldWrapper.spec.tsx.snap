// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text sub rendered properly 1`] = `
.c0 {
  display: flex;
  gap: 0.5rem;
  align-items: center;
  justify-content: space-between;
}

@media (max-width: 480px) {
  .c0 {
    flex-direction: column;
    justify-content: center;
    gap: 0.25rem;
  }
}

<div
  className="c0"
/>
`;
